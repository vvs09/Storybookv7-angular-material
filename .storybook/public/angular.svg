<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
