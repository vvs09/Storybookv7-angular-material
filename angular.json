{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}