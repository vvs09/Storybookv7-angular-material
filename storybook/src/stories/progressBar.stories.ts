import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import {
  MatProgressBarModule,
  MatProgressBar,
} from '@angular/material/progress-bar';

const meta: Meta<MatProgressBar> = {
  title: 'Components/Progress Bar',
  component: MatProgressBar,
  tags: ['autodocs'],
  argTypes: {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    bufferValue: {
      control: 'number',
      description: 'Buffer value of the progress bar. Defaults to zero.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '0' },
      },
    },
    mode: {
      control: 'radio',
      options: ['determinate', 'indeterminate', 'buffer', 'query'],
      description:
        'Mode of the progress bar.Input must be one of these values: determinate, indeterminate, buffer, query, defaults to determinate. Mirrored to mode attribute.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    value: {
      control: 'number',
      description:
        'Value of the progress bar. Defaults to zero. Mirrored to aria-valuenow.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '0' },
      },
    },
    defaultColor: {
      control: 'text',
      description: 'Default color to fall back to if no value is set.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    animationEnd: {
      control: 'object',
      description:
        'Event emitted when animation of the primary progress bar completes. This event will not be emitted when animations are disabled, nor will it be emitted for modes with continuous animations (indeterminate and query).',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '50' },
      },
    },
  },
  decorators: [
    moduleMetadata({
      imports: [MatProgressBarModule],
    }),
  ],
  render: (args: MatProgressBar) => ({
    props: args,
    template: `<mat-progress-bar ${argsToTemplate(args)} ></mat-progress-bar>`,
  }),
};

export default meta;
type Story = StoryObj<MatProgressBar>;


export const Default: Story = {
  args: {
    mode:'buffer',
    color:'accent',
    value:0,
    bufferValue:0
  },
};

export const IndeterminateProgressBar: Story = {
  args: {
    ...Default.args,
    mode:'indeterminate'
  },
};

export const DeterminateProgressBar: Story = {
  args: {
    ...Default.args,
    value:20,
    mode:'determinate',
    bufferValue:75
  },
};

export const QueryProgressBar: Story = {
  args: {
    ...Default.args,
    mode:'query',
    value:30,
    bufferValue:75
  },
};

export const PrimaryProgressBar: Story = {
  args: {
    ...Default.args,
    color:'primary',
    mode:'indeterminate',
    value:30,
    bufferValue:75
  },
};

export const WarnProgressBar: Story = {
  args: {
    ...Default.args,
    color:'warn'
  },
};
