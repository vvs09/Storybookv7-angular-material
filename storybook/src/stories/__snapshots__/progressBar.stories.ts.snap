// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Progress Bar Default smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-accent mdc-linear-progress--animation-ready"
                    ng-reflect-mode="buffer"
                    ng-reflect-color="accent"
                    ng-reflect-value="0"
                    ng-reflect-buffer-value="0"
                    aria-valuenow="0"
                    mode="buffer"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 0%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(0);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;

exports[`Components/Progress Bar DeterminateProgressBar smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-accent mdc-linear-progress--animation-ready"
                    ng-reflect-mode="determinate"
                    ng-reflect-color="accent"
                    ng-reflect-value="20"
                    ng-reflect-buffer-value="75"
                    aria-valuenow="20"
                    mode="determinate"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 100%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(0.2);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;

exports[`Components/Progress Bar IndeterminateProgressBar smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-accent mdc-linear-progress--animation-ready mdc-linear-progress--indeterminate"
                    ng-reflect-mode="indeterminate"
                    ng-reflect-color="accent"
                    ng-reflect-value="0"
                    ng-reflect-buffer-value="0"
                    mode="indeterminate"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 100%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(1);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;

exports[`Components/Progress Bar PrimaryProgressBar smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-primary mdc-linear-progress--animation-ready mdc-linear-progress--indeterminate"
                    ng-reflect-mode="indeterminate"
                    ng-reflect-color="primary"
                    ng-reflect-value="30"
                    ng-reflect-buffer-value="75"
                    mode="indeterminate"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 100%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(1);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;

exports[`Components/Progress Bar QueryProgressBar smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-accent mdc-linear-progress--animation-ready mdc-linear-progress--indeterminate"
                    ng-reflect-mode="query"
                    ng-reflect-color="accent"
                    ng-reflect-value="30"
                    ng-reflect-buffer-value="75"
                    mode="query"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 100%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(1);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;

exports[`Components/Progress Bar WarnProgressBar smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-progress-bar role="progressbar"
                    aria-valuemin="0"
                    aria-valuemax="100"
                    tabindex="-1"
                    class="mat-mdc-progress-bar mdc-linear-progress mat-warn mdc-linear-progress--animation-ready"
                    ng-reflect-mode="buffer"
                    ng-reflect-color="warn"
                    ng-reflect-value="0"
                    ng-reflect-buffer-value="0"
                    aria-valuenow="0"
                    mode="buffer"
  >
    <div aria-hidden="true"
         class="mdc-linear-progress__buffer"
    >
      <div class="mdc-linear-progress__buffer-bar"
           style="flex-basis: 0%;"
      >
      </div>
      <div class="mdc-linear-progress__buffer-dots">
      </div>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
         style="transform: scaleX(0);"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
    <div aria-hidden="true"
         class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span class="mdc-linear-progress__bar-inner">
      </span>
    </div>
  </mat-progress-bar>
</storybook-root>
`;
